### PR TITLE
Remove vendor proto files and use opencensus-proto instead.

### LINKED ISSUE #318: Move proto-generated files to OpenCensus-Proto.
I'm new to the dependency management in Python so this may be a naive question:

For the OC-Agent Exporter, we put the proto-generated Python files as a vendor under [opencensus/trace/exporters/gen/opencensus](https://github.com/census-instrumentation/opencensus-python/tree/master/opencensus/trace/exporters/gen/opencensus). AFAIS this is not convenient since every time when we made a new release on [OpenCensus-Proto](https://github.com/census-instrumentation/opencensus-proto), those gen-files need to be updated separately. This has also caused some trouble with Pylint before (https://github.com/census-instrumentation/opencensus-python/pull/266#issuecomment-418474228).

If possible, I think a better way is to host the generated Python files on [OpenCensus-Proto](https://github.com/census-instrumentation/opencensus-proto), release them as separate Python library, and have the OC-Agent Exporter import it. Note that Go [already did it](https://github.com/census-instrumentation/opencensus-proto/tree/master/gen-go).

/cc @lmolkova @liyanhui1228

### PR DESCRIPTION
**Pull request by songy23**
~~DO NOT MERGE until https://github.com/census-instrumentation/opencensus-proto/pull/197 is merged and a new release is cut.~~

Fixes #318.

### CODE REVIEW

File: contrib/opencensus-ext-ocagent/setup.py [hunk 1]
```diff
@@ -41,6 +41,7 @@
     install_requires=[
         'grpcio >= 1.0.0, < 2.0.0',
         'opencensus >= 0.4.dev0, < 1.0.0',
+        'opencensus-proto >= 0.0.dev0, < 1.0.0',
```
Comment (c24t) [inline contrib/opencensus-ext-ocagent/setup.py]: FYI we've been replacing "dev" in the release branches. If `opencensus-proto` follows the same versioning strategy the first release will be `0.1.0` and the version number on master after the release should be `0.2.dev0`, in which case this should be `0.1.dev0`.

**Comment by reyang** [comment]
@songy23 just realized there was a lint rule which we can get rid off. You can leave it to me since we'll move from flake8 to pylint, up to you.

https://github.com/census-instrumentation/opencensus-python/blob/master/nox.py#L119

**Comment by songy23** [comment]
Got it, I'll leave it to you then because I'm putting this PR on hold until we made the next release for opencensus-proto (currently it's 0.0.dev0).

**Comment by songy23** [comment]
https://github.com/census-instrumentation/opencensus-proto/pull/197 is merged, now I think we can move forward with this PR.

**Pull request merged**